opslevel/resource_opslevel_team.go

```diff
@@ -195,40 +207,69 @@ func (teamResource *TeamResource) Read(ctx context.Context, req resource.ReadReq
 		return
 	}
 
-	readTeamResourceModel, diags := NewTeamResourceModel(ctx, *team)
+	newStateModel, diags := newTeamResourceModel(ctx, *team, stateModel.Parent.ValueString())
 	resp.Diagnostics.Append(diags...)
-	// if parent is set, use an ID or alias for this field based on what is currently in the state
-	if opslevel.IsID(data.Parent.ValueString()) {
-		readTeamResourceModel.Parent = types.StringValue(string(team.ParentTeam.Id))
-	} else {
-		// TODO: error thrown if config has alias from the parent team that is not the default alias
-		readTeamResourceModel.Parent = OptionalStringValue(team.ParentTeam.Alias)
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	resp.Diagnostics.Append(resp.State.Set(ctx, &readTeamResourceModel)...)
+	removeNonTerraformManagedMembers(ctx, &newStateModel, stateModel)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &newStateModel)...)
 }
 
 func (teamResource *TeamResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	var data TeamResourceModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	var planModel teamResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	members, err := getMembers(data.Member)
+	// need to read current state model so that members can be added/removed
+	var stateModel teamResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// for each member that is in the state, but not in the plan, remove that member from the team
+	membersToDelete := make([]opslevel.TeamMembershipUserInput, 0)
+	for _, mem := range stateModel.Member {
+		if !slices.Contains(planModel.Member, mem) {
+			membersToDelete = append(membersToDelete, opslevel.TeamMembershipUserInput{
+				User: opslevel.NewUserIdentifier(mem.Email.ValueString()),
+				Role: mem.Role.ValueStringPointer(),
+			})
+		}
+	}
+	_, err := teamResource.client.RemoveMemberships(&opslevel.TeamId{Id: opslevel.ID(planModel.Id.ValueString())}, membersToDelete...)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to delete members, got error: %s", err))
+		return
+	}
+	// for each member that is in the plan, add that member to the team
+	// by "adding" a member, you can also change someone from being a contributor to manager and vice versa.
+	// therefore, this step should be performed AFTER removing members.
+	membersToAdd := make([]opslevel.TeamMembershipUserInput, len(planModel.Member))
+	for i, mem := range planModel.Member {
+		membersToAdd[i] = opslevel.TeamMembershipUserInput{
+			User: opslevel.NewUserIdentifier(mem.Email.ValueString()),
+			Role: mem.Role.ValueStringPointer(),
+		}
+	}
+	_, err = teamResource.client.AddMemberships(&opslevel.TeamId{Id: opslevel.ID(planModel.Id.ValueString())}, membersToAdd...)
```

Comment:
would have been nice to break this logic up into discrete functions that take input data and output data to pass into `RemoveMemberships` and `AddMemberships`.  This way we could put those discrete functions under unit test with different input cases via a table test to ensure the logic continue to work as expected.

Reply:
Added ticket to do this - https://github.com/OpsLevel/team-platform/issues/366